lib/srv/alpnproxy/gcp_middleware.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alpnproxy
+
+import (
+	"crypto/subtle"
+	"net/http"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+// GCPMiddleware implements a simplified version of MSI server serving auth tokens.
```

Comment:
I think that **MSI** is Azure specific name and doesn't apply to Google Cloud

Reply:
Fixed docs here and elsewhere.